### PR TITLE
Feature/OSCBS-1-bacnet-secure-connect

### PR DESCRIPTION
**Pull request by cybermag**
Added BACnet/SC using lib-websockets and ported to the command line tools on Windows, Linux, MacOS, and BSD.  
Added an example BACnet/SC hub application.

### CODE REVIEW

File: src/bacnet/basic/sys/fifo.h [hunk 1]
```diff
@@ -75,6 +75,13 @@ extern "C" {
     BACNET_STACK_EXPORT
     uint8_t FIFO_Peek(
         FIFO_BUFFER const *b);
+    BACNET_STACK_EXPORT
+
+    BACNET_STACK_EXPORT
```
Comment (skarg) [inline src/bacnet/basic/sys/fifo.h:80]: please remove extra BACNET_STACK_EXPORT

File: src/bacnet/hostnport.c [hunk 2]
```diff
@@ -245,6 +245,32 @@ int host_n_port_decode(uint8_t *apdu,
     return apdu_len;
 }
 
+int host_n_port_context_decode(uint8_t *apdu,
```
Comment (skarg) [inline src/bacnet/hostnport.c:248]: please pass apdu_size and use secure decoding functions.

**Review by skarg — CHANGES_REQUESTED** [review]

File: src/bacnet/config.h [hunk 3]
```diff
@@ -102,6 +104,27 @@
...
+#endif
+#ifndef SC_NETPORT_NPDU_MAX
+#define SC_NETPORT_NPDU_MAX 1500
+#endif
+#ifndef SC_NETPORT_CONNECT_TIMEOUT
+#define SC_NETPORT_CONNECT_TIMEOUT 5
+#endif
+#ifndef SC_NETPORT_HEARTBEAT_TIMEOUT
+#define SC_NETPORT_HEARTBEAT_TIMEOUT 3
+#endif
```
Comment (stargieg) [inline src/bacnet/config.h:119]: The Heartbeat should be 60 s.
Tested with Loytec (Cimetrics BACstac runtime v6.4.2.6).
If the Heartbeat is greater than 150s (multiply 2 in bsc_process_srv_awaiting_request() 300s) than
crash the Cimetrics BACstac after 300s. Other People Problem.

But in bsc-socket.h is
```
    /* According 12.56.Y10 SC_Heartbeat_Timeout */
    /* (http://www.bacnet.org/Addenda/Add-135-2020cc.pdf) the recommended default */
    /* value is 300 seconds. */
```
The default is 300s. In this case SC_NETPORT_DISCONNECT_TIMEOUT should be 150s.

**Comment by skarg** [comment]
Changed branch name to [feature/bacnet-secure-connect-hub](https://github.com/bacnet-stack/bacnet-stack/pull/new/feature/bacnet-secure-connect-hub)

**Pull request closed without merging**